src/oncall/scheduler/round-robin.py

```diff
@@ -0,0 +1,113 @@
+from datetime import datetime, timedelta
+from pytz import timezone, utc
+from falcon import HTTPBadRequest
+import time
+import default
+import logging
+
+logger = logging.getLogger()
+
+class Scheduler(default.Scheduler):
+
+    def guess_last_scheduled_user(self, schedule, start, roster, cursor):
+        cursor.execute('''
+                        SELECT `user_id`, MAX(`end`) AS `last_end` FROM `event`
+                        WHERE `team_id` = %s AND `user_id` IN %s AND `end` <= %s
+                        AND `role_id` = %s
+                        GROUP BY `user_id`
+                        ''', (schedule['team_id'], roster, start, schedule['role_id']))
+        if cursor.rowcount != 0:
+            # Grab user id with largest last scheduled time
+            return max(cursor.fetchall(), key=lambda x: x['last_end'])['user_id']
```

Comment:
I might be wrong but could this be done in SQL itself to avoid needing to put that max/lambda against all of the rows returned?